numpy/linalg/umath_linalg.c.src

```diff
@@ -68,15 +68,6 @@ dbg_stack_trace()
 
 typedef CBLAS_INT         fortran_int;
 
-typedef struct { float r, i; } f2c_complex;
-typedef struct { double r, i; } f2c_doublecomplex;
-/* typedef long int (*L_fp)(); */
-
-typedef float             fortran_real;
-typedef double            fortran_doublereal;
-typedef f2c_complex       fortran_complex;
-typedef f2c_doublecomplex fortran_doublecomplex;
```

Comment:
It would be good to understand why these typedefs were originally used.

Reply:
I think they were probably copied from f2c. Similar typedefs appear in f2c.h in the lapack_lite folder.

Reply:
That's a convincing enough argument for me:

https://github.com/numpy/numpy/blob/beac56601587e26563d83b0a47cd0b1f006bf5d6/numpy/linalg/lapack_lite/f2c.h#L19-L22